internal/runner/rspec.go

```diff
@@ -16,19 +16,13 @@ import (
 	"github.com/kballard/go-shellquote"
 )
 
-// In future, Rspec will implement an interface that defines
-// behaviour common to all test runners.
-// For now, Rspec provides rspec specific behaviour to execute
-// and report on tests in the Rspec framework.
+var _ = TestRunner(Rspec{})
+
 type Rspec struct {
-	TestCommand            string
-	TestFileExcludePattern string
-	TestFilePattern        string
-	RetryTestCommand       string
-	ResultPath             string
+	RunnerConfig
 }
 
-func NewRspec(r Rspec) *Rspec {
+func NewRspec(r RunnerConfig) Rspec {
```

Comment:
I don't know much Go, but how come this has changed to return a struct rather than a pointer?

Reply:
I had a wee pairing session with @wolfeidau on this last week. My recollection (forgive me if this is wrong Mark!) is that since we're not mutating the returned value in any way and there is no shared state between Rspec (or jest) instances - since there is only one - it's neater to just return the value rather than the pointer so we can avoid de-referencing it everywhere we need to use it.

Reply:
@wooly correct, this is more of a "configuration" or immutable thing, not a stateful thing. 

As mentioned this can trip people up so most certainly something to keep an eye on.

Reply:
Ah, okay cool!